src/app/clusters/ota-requestor/ota-requestor.h

```diff
@@ -0,0 +1,109 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/* This file contains the declarations for the Matter OTA Requestor implementation and API.
+ *  Applications and platforms implementing the OTA Requestor functionality must include this
+ * file
+ */
+
+#pragma once
+
+// Interface class to abstract the OTA-related business logic. Each application
+// must implement this interface. All calls must be non-blocking unless stated otherwise
+class OTARequestorDriver {
+public:
+    // A call into the application logic to give it a chance to allow or stop the Requestor
+    // from proceeding with actual image download.
+    virtual bool AllowImageDownload() = 0;
```

Comment:
Should this be `HandleImageDownload()` or something to that effect? Given that it could be the case the requestor could stop the image download, "Allow" sounds misleading

Reply:
"Allow" is the purpose behind this API. I've updated the comments to further clarify this.

Reply:
Updated the name